autometa/common/utilities.py

```diff
@@ -402,124 +406,139 @@ def write_checksum(infpath, outfpath):
+        if not os.path.exists(fpath):
+            raise FileNotFoundError(fpath)
+        checksum_hash, checksum_name = calc_checksum(fpath).split()
+        checksum_timestamp = datetime.now().timestamp()
+        checksum_time = datetime.fromtimestamp(checksum_timestamp).strftime(
+            "%Y-%m-%d_%H-%M-%S"
+        )
+        atimestamp = os.path.getatime(fpath)
+        accession_time = datetime.fromtimestamp(atimestamp).strftime(
+            "%Y-%m-%d_%H-%M-%S"
+        )
+        mtimestamp = os.path.getmtime(fpath)
+        modified_time = datetime.fromtimestamp(mtimestamp).strftime("%Y-%m-%d_%H-%M-%S")
+        checkpoint = Checkpoint(
+            hash=checksum_hash,
+            name=checksum_name,
+            filepath=os.path.realpath(fpath),
+            accession_time=accession_time,
+            modified_time=modified_time,
+            checksum_time=checksum_time,
+        )
+        checkpoints.append(checkpoint)
+    if dtype == "list":
+        return checkpoints
+    else:
+        return pd.DataFrame(checkpoints)
+
+
+def merge_checkpoints(old_checkpoint_fpath, new_checkpoints, overwrite=False):
+    """Merge existing checkpoints with new checkpoints
 
     Parameters
     ----------
-    checkpoint_fp : str
-        </path/to/checkpoints.tsv>
-    fpath : str
-        </path/to/file>
+    old_checkpoint_fpath : str
+        Path to existing checkpoints file
+    new_checkpoints : iterable
+        list of files to be made into checkpoints
+    overwrite : bool, optional
+        Will overwrite `old_checkpoint_fpath` after `new_checkpoints` have been merged, by default False
 
     Returns
     -------
-    dict
-        {fp:checksum, ...}
-
+    pd.DataFrame
+        Checkpoints dataframe
+        index=enumerated cols=["hash","name","filepath","accession_time","modified_time"]
     """
-    checkpoints = get_checkpoints(checkpoint_fp)
-    if valid_checkpoint(checkpoint_fp, fpath):
-        return checkpoints
-    new_checksum = calc_checksum(fpath)
-    checkpoints.update({fpath: new_checksum})
-    outlines = ""
-    for fp, chk in checkpoints.items():
-        outlines += f"{chk}\t{fp}\n"
-    with open(checkpoint_fp, "w") as fh:
-        fh.write(outlines)
-    logger.debug(f"Checkpoints updated: {new_checksum[:16]} {os.path.basename(fpath)}")
-    return checkpoints
+    old_df = get_existing_checkpoints(fpath=old_checkpoint_fpath)
+    new_df = make_inputs_checkpoints(inputs=new_checkpoints)
+    df = pd.concat([old_df, new_df]).drop_duplicates(["hash", "name", "filepath"])
+    df.reset_index(drop=True)
+    if overwrite:
+        df.to_csv(old_checkpoint_fpath, sep="\t", index=False, header=True)
+    return df
+
+
+def is_checkpoint(output):
+    if not os.path.getsize(output):
+        return False
+    dirpath = os.path.dirname(os.path.realpath(output))
+    checkpoints_fpath = os.path.join(dirpath, "checkpoints.tsv")
+    checkpoints_df = get_existing_checkpoints(checkpoints_fpath)
```

Comment:
I am unclear what this function is doing or why we need these three lines of code when nothing that is done here is returned. It seems that the `get_existing_checkpoints` function will only cause an error when `checkpoints_fpath` isn't a string, so running it here does not seem to be any sort of check that might stop `return True`.